### ISSUE TITLE
Key passphrase dialog is sometimes inaccessible

### ISSUE THREAD

**Issue by akater** [ecraven/pinentry-emacs#7]
Sometimes pinentry-emacs dialog won't appear, even though emacs-pinentry process is started. I'm not sure if this is pinentry-emacs related, as the bug is also not easy to reproduce.

I use [EXWM](https://github.com/ch11ng/exwm) but had this issue in standalone emacs too, though it seems EXWM still could be partially responsible. Still, even in standalone emacs the bug (in a less severe form) routinely occurs on `M-x find-file /sudo::/etc/ TAB` — at this moment tramp requests password from [password-store](https://www.passwordstore.org/). No dialog appears on screen, only message “Tramp: Sending password”. In this particular situation, `C-g` makes the passphrase entry dialog to appear, although sometimes stripped to its last line only (this last peculiarity could have something to do with my echo area being hidden by default). Note: this workaround is far less robust for `M-x find-file /sudo::/ TAB`, without `etc`: is the latter case, on `C-g` tramp often simply complains it can't obtain password and quits. This looks more like a tramp issue but might provide some insight.

Sometimes, though, the dialog is inaccessible when I request arbitrary passwords from password-store by means of `password-store-copy` function from package password-store, in which cases Emacs freezes for a minute or so (I assume pinentry-emacs process is then terminated due to user inactivity). It seems 99% safe to call `pass -c`, even from external terminal run under EXWM control (the dialog is likely always accessible this way), certainly safer than by means of `password-store-copy`.

Overall, it could be password-store-specific issue. Tramp probably uses [auth-password-store](https://github.com/DamienCassou/auth-password-store) which in turn uses password-store. However, the symptoms are different, and I have vague memories of the issue being present independently of password-store. After spending some weeks in doubt, I decided to post here. 

All packages are from melpa — not from melpa-stable but the issue is persistent for some months, since 2016-09, I guess.

Emacs 24, GTK+2 build

**Comment by dieggsy** [comment]
I don't think this is pinentry-emacs related, as i'm having a similar issue and not using pinentry-emacs.

**Comment by ecraven** [comment]
I've started using exwm a few days ago, and had the problem that the gtk pinentry would not let me enter any actual text (though the buttons and checkbox worked fine).
I've updated pinentry-emacs to work for me (current arch linux), do you still see these problems?

**Comment by akater** [comment]
On Emacs 25, I'm using pinentry-emacs + exwm with almost no
issues. Looks like this comment [1] is exhaustive.

If another emacs is run in terminal and was started after the gui emacs,
then pinentry window would appear there, so that's another way to live
with it.

I have some-shortcut for password-store-copy, so in the end I basically
some-shorctut C-m C-g some-shortcut query C-m passphrase
instead of
some-shortcut query C-m passphrase

Even with this little annoyance I'd pick such a workflow over gtk/qt
window any time. I don't know how to make tty/ncurses pinentries
work at all, and don't bother atm.

[1] https://github.com/ch11ng/exwm/issues/288#issuecomment-320468536
